### PR TITLE
Generalize Eidos bio processing

### PR DESCRIPTION
**Pull request by bgyori**
This PR refactors bio-oriented processing of extractions from Eidos and allows using a custom grounder to add groundings for agents.

### CODE REVIEW

File: indra/sources/eidos/bio_processor.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        name = norm_txt
+    # We take whatever grounding and name are available and then
+    # standardize the agent.
+    agent = Agent(name, db_refs={'TEXT_NORM': norm_txt,
+                                 'TEXT': raw_txt, **gr})
+    standardize_agent_name(agent, standardize_refs=True)
+    return agent
+
+
+def default_grounder_wrapper(text, context):
```
Comment (cthoyt) [inline indra/sources/eidos/bio_processor.py:69]: can we do some typing for this? `(text: str, context: Optional[str]) -> Tuple[Mapping[str, Any], List[Any]]`

Comment (bgyori) [reply indra/sources/eidos/bio_processor.py:69]: yes, do you want to add those and push to my branch?

File: indra/sources/eidos/bio_processor.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+from indra.statements import Activation, Inhibition
+from indra.ontology.standardize import standardize_agent_name
+from indra.preassembler.grounding_mapper.gilda import get_grounding
+from .processor import EidosProcessor
+
+
+class EidosBioProcessor(EidosProcessor):
+    """Class to extract biology-oriented INDRA statements from Eidos output
+    in a way that agents are grounded to biomedical ontologies."""
+    def __init__(self, json_dict, grounder=None):
```
[the remark this answers is not in the document]
Comment (cthoyt) [reply indra/sources/eidos/bio_processor.py:11]: These type hints are getting pretty wild, but they definitely help the IDE and new devs understand what's going on

[the remark this answers is not in the document]
Comment (cthoyt) [reply indra/sources/eidos/bio_processor.py:11]: Plus you can import this type hint and stick it everywhere throughout this PR

File: indra/sources/eidos/bio_processor.py (hunk 1 above)
Comment (cthoyt) [reply indra/sources/eidos/bio_processor.py:69]: So PR to your PR?

Comment (cthoyt) [reply indra/sources/eidos/bio_processor.py:69]: Otherwise you have to give me push rights on bgyori/indra

File: indra/tests/test_eidos.py [hunk 3]
```diff
@@ -283,6 +283,15 @@ def test_get_agent_bio():
         assert agent.db_refs['TEXT_NORM'] == norm_text
 
 
+def test_bio_custom_grounding():
+    def my_grounder(txt, context):
+        return {'MYDB': 'MYGROUNDING'}
```
Comment (cthoyt) [inline indra/tests/test_eidos.py:288]: are these dictionaries always `Mapping[str, str]`? That would improve the type annotation over `Mapping[str, Any]`

**Pull request merged**